src/components/input/NumericInput.js

```diff
@@ -95,6 +95,21 @@ class NumericInput extends PureComponent {
     clearInterval(this.timer.current);
   };
 
+  // Unlike mouse down events, key down events are "auto-repeated" while holding down the key
+  // This means we don't need to use an interval and can directly rely on the handler
+  // https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent#auto-repeat_handling
+  handleKeyDown = (event) => {
+    if (event.key === 'ArrowUp') {
```

Comment:
PP: prevent using plain strings in the code. e.g. introduce a `Key` file/object so you can do `if(event.key === Key.ARROW_UP)`